cli/azd/pkg/commands/pipeline/azdo_provider.go

```diff
@@ -7,65 +7,528 @@ import (
+	if err != nil {
+		return "", fmt.Errorf("error saving project name to environment %w", err)
+	}
+	var remoteUrl string
+
+	if !newProject {
+		remoteUrl, err = p.promptForAzdoRepository(ctx, console)
+		if err != nil {
+			return "", err
+		}
+	} else {
+		remoteUrl, err = p.getDefaultRepoRemote(ctx, projectName, projectId, console)
+		if err != nil {
+			return "", err
+		}
+	}
+	return remoteUrl, nil
+}
+
+// returns the git remote for a newly created repo that is part of a newly created AzDo project
+func (p *AzdoScmProvider) getDefaultRepoRemote(ctx context.Context, projectName string, projectId string, console input.Console) (string, error) {
+	connection, err := p.getAzdoConnection(ctx)
+	if err != nil {
+		return "", err
+	}
+	repo, err := azdo.GetDefaultGitRepositoriesInProject(ctx, projectName, connection)
+	if err != nil {
+		return "", err
+	}
+
+	err = p.StoreRepoDetails(ctx, repo)
+	if err != nil {
+		return "", err
+	}
+
+	message := fmt.Sprintf("using default repo (%s) in newly created project(%s)", *repo.Name, projectName)
+	console.Message(ctx, message)
+
+	return *repo.RemoteUrl, nil
+}
+
+// prompt the user to select azdo repo or create a new one
+func (p *AzdoScmProvider) promptForAzdoRepository(ctx context.Context, console input.Console) (string, error) {
+	var remoteUrl string
+	// There are a few ways to configure the remote so offer a choice to the user.
+	idx, err := console.Select(ctx, input.ConsoleOptions{
+		Message: fmt.Sprintf("How would you like to configure your remote? (Organization: %s)", p.repoDetails.projectName),
+		Options: []string{
+			"Select an existing Azure DevOps Repository",
+			"Create a new private Azure DevOps Repository",
+		},
+		DefaultValue: "Create a new private Azure DevOps Repository",
+	})
+
+	if err != nil {
+		return "", fmt.Errorf("prompting for remote configuration type: %w", err)
+	}
+
+	switch idx {
+	// Select from an existing Azure Devops project
+	case 0:
+		remoteUrl, err = p.ensureGitRepositoryExists(ctx, console)
+		if err != nil {
+			return "", err
+		}
+
+	// Create a new project
+	case 1:
+		remoteUrl, err = p.createNewGitRepositoryFromInput(ctx, console)
+		if err != nil {
+			return "", err
+		}
+
+	default:
+		panic(fmt.Sprintf("unexpected selection index %d", idx))
+	}
+
+	return remoteUrl, nil
+}
+
+// defines the structure of an ssl git remote
+var azdoRemoteGitUrlRegex = regexp.MustCompile(`^git@ssh.dev.azure\.com:(.*?)(?:\.git)?$`)
+
+// defines the structure of an HTTPS git remote
+var azdoRemoteHttpsUrlRegex = regexp.MustCompile(`^https://[a-zA-Z0-9]+(?:-[a-zA-Z0-9]+)*:*.+@dev.azure\.com/(.*?)$`)
+
+// ErrRemoteHostIsNotAzDo the error used when a non Azure DevOps remote is found
+var ErrRemoteHostIsNotAzDo = errors.New("existing remote is not an Azure DevOps host")
+
+// helper function to determine if the provided remoteUrl is an azure devops repo.
+// currently supports AzDo PaaS
+func isAzDoRemote(remoteUrl string) error {
```

Comment:
You might be able to remove slug here (since we don't end up using it) and write something more like:

```
func isAzDoRemote(remoteUrl string) error {
	for _, r := range []*regexp.Regexp{azdoRemoteGitUrlRegex, azdoRemoteHttpsUrlRegex} {
		if r.MatchString(remoteUrl) {
			return nil
		}
	}

	return ErrRemoteHostIsNotAzDo
}
```

Reply:
https://github.com/Azure/azure-dev/issues/756